plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java

```diff
@@ -0,0 +1,411 @@
+     * when no input mappings provided, default to add all fields to model input fields,
+     * when no output mapping provided, default to output as
+     * "inference_results" field (the same format as predict API)
+     * @param ingestDocument The IngestDocument object containing the data to be processed.
+     * @param batchPredictionListener The GroupedActionListener for batch prediction.
+     * @param processInputMap A list of maps containing input field mappings.
+     * @param processOutputMap A list of maps containing output field mappings.
+     * @param i The current index of the prediction task.
+     * @param num_of_prediction_tasks The total number of prediction tasks.
+     */
+    private void processPredictions(
+        IngestDocument ingestDocument,
+        GroupedActionListener<Void> batchPredictionListener,
+        List<Map<String, String>> processInputMap,
+        List<Map<String, String>> processOutputMap,
+        int i,
+        int num_of_prediction_tasks
+    ) {
+        Map<String, String> modelParameters = new HashMap<>();
+        if (inferenceProcessorAttributes.getModelConfigMaps() != null) {
+            modelParameters.putAll(inferenceProcessorAttributes.getModelConfigMaps());
+        }
+        // when no input mapping is provided, default to read all fields from documents as model input
+        if (num_of_prediction_tasks == 0) {
+            Set<String> documentFields = ingestDocument.getSourceAndMetadata().keySet();
+            for (String field : documentFields) {
+                getMappedModelInputFromDocuments(ingestDocument, modelParameters, field, field);
+            }
+
+        } else {
+            Map<String, String> inputMapping = processInputMap.get(i);
+            for (Map.Entry<String, String> entry : inputMapping.entrySet()) {
+                // model field as key, document field as value
+                String originalFieldName = entry.getValue();
+                String modelInputFieldName = entry.getKey();
+                getMappedModelInputFromDocuments(ingestDocument, modelParameters, originalFieldName, modelInputFieldName);
+            }
+        }
+
+        ActionRequest request = getRemoteModelInferenceRequest(modelParameters, inferenceProcessorAttributes.getModelId());
+
+        client.execute(MLPredictionTaskAction.INSTANCE, request, new ActionListener<>() {
+
+            @Override
+            public void onResponse(MLTaskResponse mlTaskResponse) {
+                ModelTensorOutput modelTensorOutput = (ModelTensorOutput) mlTaskResponse.getOutput();
+                if (processOutputMap == null || processOutputMap.isEmpty()) {
+                    appendFieldValue(modelTensorOutput, null, DEFAULT_OUTPUT_FIELD_NAME, ingestDocument);
+                } else {
+                    Map<String, String> outputMapping = processOutputMap.get(i);
+
+                    for (Map.Entry<String, String> entry : outputMapping.entrySet()) {
```

Comment:
Add some comments to explain if not configured in output mapping, we will not write the result to index.